aws/cf_template.go

```diff
@@ -112,11 +115,13 @@ func generateTemplate(spec *stackSpec) (string, error) {
 		}
 	}
 
-	protocol := "HTTP"
-	tlsProtocol := "HTTPS"
+	protocol := httpProtocol
+	tlsProtocol := httpsProtocol
 	if spec.loadbalancerType == LoadBalancerTypeNetwork {
 		protocol = "TCP"
 		tlsProtocol = "TLS"
+	} else if spec.targetHTTPS {
+		protocol = httpsProtocol
```

Comment:
This changes protocol of the targetGroup but also protocol of the _HTTP_ listener:
https://github.com/zalando-incubator/kube-ingress-aws-controller/blob/92fa334e0d6332d34586aa226d2b2b00dc3453d0/aws/cf_template.go#L159

If I am not mistaken HTTP Listener protocol should be either "HTTP" or "TCP" in case of NLB

Reply:
I have created https://github.com/zalando-incubator/kube-ingress-aws-controller/pull/434 if that makes sense